com.unity.shadergraph/Editor/Data/Graphs/DynamicVectorMaterialSlot.cs

```diff
@@ -21,7 +21,7 @@ class DynamicVectorMaterialSlot : MaterialSlot, IMaterialSlotHasValue<Vector4>
 
         static readonly string[] k_Labels = {"X", "Y", "Z", "W"};
 
-        private ConcreteSlotValueType m_ConcreteValueType = ConcreteSlotValueType.Vector4;
```

Comment:
Could you explain why this change was needed?

Reply:
Also, will this give any weird behavior change to our math nodes and the like?

Reply:
The slot was behaving inconsistently. Initially it would show a 4 channel initializer (0,0,0,0). If you then connected to it (any node), and then disconnected again, the slot would turn into a single channel initializer. With this change it always shows a default single channel initializer, so you don't have that disparity.
Also worth noting that other nodes that accept a dynamic vector, do already use a single channel initializer, so this was notably the odd one out.